bindings/wasm/examples/src/1_advanced/6_domain_linkage.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Client, MnemonicSecretManager } from "@iota/client-wasm/node";
+import { Bip39 } from "@iota/crypto.js";
+import {
+    Credential,
+    CredentialValidationOptions,
+    DIDUrl,
+    DomainLinkageConfiguration,
+    DomainLinkageValidator,
+    Duration,
+    IotaDID,
+    IotaDocument,
+    IotaIdentityClient,
+    LinkedDomainService,
+    ProofOptions,
+    Timestamp,
+} from "@iota/identity-wasm/node";
+import { IAliasOutput, IRent, TransactionHelper } from "@iota/iota.js";
+import { API_ENDPOINT, createDid } from "../util";
+
+/**
+ * Demonstrates how to link a domain and a DID and verify the linkage.
+ */
+export async function domainLinkage() {
+    const client = new Client({
+        primaryNode: API_ENDPOINT,
+        localPow: true,
+    });
+    const didClient = new IotaIdentityClient(client);
+
+    // Generate a random mnemonic for our wallet.
+    const secretManager: MnemonicSecretManager = {
+        mnemonic: Bip39.randomMnemonic(),
+    };
+
+    // Creates a new wallet and identity (see "0_create_did" example).
+    let { document, keypair } = await createDid(client, secretManager);
+    const did: IotaDID = document.id();
+
+    // =====================================================
+    // Create Linked Domain service
+    // ====================================================
+
+    let domainFoo = "https://foo.example.com";
+    let domainBar = "https://bar.example.com";
+
+    // Create a Linked Domain Service to enable the discovery of the linked domains through the DID Document.
+    // This is optional
```

Comment:
Why is it optional? I'd say either remove or explain.

Reply:
It means that the specs does not require the DID Document to have a service for correct domain linkage.
I edited the comment.